src/wp-content/themes/twentytwentyone/assets/js/primary-navigation.js

```diff
@@ -173,17 +173,19 @@ function twentytwentyoneExpandSubMenu( el ) { // jshint ignore:line
 		 * @since Twenty Twenty-One 1.1
 		 */
 		document.addEventListener( 'click', function( event ) {
-			// If target onclick is <a> with # within the href attribute
-			if ( event.target.hash && event.target.hash.includes( '#' ) ) {
-				wrapper.classList.remove( id + '-navigation-open', 'lock-scrolling' );
-				twentytwentyoneToggleAriaExpanded( mobileButton );
-				// Wait 550 and scroll to the anchor.
-				setTimeout(function () {
-					var anchor = document.getElementById(event.target.hash.slice(1));
-					if ( anchor ) {
-						anchor.scrollIntoView();
-					}
-				}, 550);
+			if ( document.getElementById( 'site-navigation' ).contains( event.target ) ) {
+				// If target onclick is <a> with # within the href attribute
+				if ( event.target.hash && event.target.hash.includes( '#' ) ) {
+					wrapper.classList.remove( id + '-navigation-open', 'lock-scrolling' );
+					twentytwentyoneToggleAriaExpanded( mobileButton );
+					// Wait 550 and scroll to the anchor.
+					setTimeout(function () {
+						var anchor = document.getElementById(event.target.hash.slice(1));
+						if ( anchor ) {
+							anchor.scrollIntoView();
+						}
+					}, 550);
```

Comment:
550 ms seems to me a magic number. Is there any reason why we use this value? If there is one, it would be nice to clarify it in an inline comment otherwise maybe use a different value that sets a reasonable delay.

Reply:
I only tried to adress the issue, this part of the code has not been updated.
You would need to look through the previous commit history to see if there is any discussions around it.

Reply:
Looks like the 550 ms delay comes from code adapted from Twenty Twenty, which in turns comes from a patch suggested on a support thread for another theme. There's no explanation about the chosen value 550. Relevant links:

https://core.trac.wordpress.org/ticket/52006
https://core.trac.wordpress.org/ticket/48916
https://github.com/WordPress/wordpress-develop/pull/254
https://wordpress.org/support/topic/mobile-menu-doesnt-work-for-anchor-links/#post-12487136

Reply:
Thank you for doing all this research!